### PR TITLE
(BKR-1382) Add the redhatfips7-64 platform

### PR DESCRIPTION
**Pull request by ScottGarman**

### CODE REVIEW

**Comment by ScottGarman** [comment]
I think I'd like to hold off on merging this until the redhat-fips-7-x86_64 template is running in the pooler, just in case it's name ends up changing.

**Comment by ScottGarman** [comment]
@jsane FYI ^

File: lib/beaker-hostgenerator/data.rb [hunk 1]
```diff
@@ -623,6 +623,15 @@ def osinfo
             'template' => 'redhat-7-x86_64'
           }
         },
+        'redhatfips7-64' => {
+          :general => {
+            'platform'           => 'redhat-fips-7-x86_64',
```
Comment (kevpl) [inline lib/beaker-hostgenerator/data.rb]: @ScottGarman have you tried a beaker run using this platform? I'd be surprised if it didn't fail, as beaker's Platform object assumes that the platform string can be nicely split on `-` characters in this pattern
```
<os_name>-<os_version>-<arch>
```

If you'd like to have a 2-part `os_name` section of the platform string, I would suggest using an `_` like [cisco nexus in this example](https://github.com/puppetlabs/beaker-hostgenerator/blob/master/lib/beaker-hostgenerator/data.rb#L201).

Comment (ScottGarman) [reply lib/beaker-hostgenerator/data.rb]: Very useful feedback @kevpl, thanks!

**Comment by ScottGarman** [comment]
@kevpl will this also require a change in beaker itself? E.g: https://github.com/puppetlabs/beaker/blob/master/lib/beaker/platform.rb#L6

**Comment by ScottGarman** [comment]
I've created a beaker PR in case this is also needed: https://github.com/puppetlabs/beaker/pull/1484

Note: I'm in a training class for the rest of today, so someone else may need to push this forward in my absence.

**Pull request merged**